packages/block-library/src/image/edit.js

```diff
@@ -371,9 +366,15 @@ class ImageEdit extends Component {
 						<SelectControl
 							label={ __( 'Image Size' ) }
 							value={ url }
-							options={ map( imageSizes, ( size, slug ) => ( {
-								value: size.source_url,
-								label: availableImageSizesBySlug[ slug ] ? availableImageSizesBySlug[ slug ].name : startCase( slug ),
+							options={ compact( map( availableImageSizes, ( { name, slug } ) => {
```

Comment:
This logic is getting more complex (compact, map, etc) and we or a plugin may also need this logic, would it make sense to move this to a memoized selector? This would make sure the logic is reusable avoid future bugs in this logic, and that we don't always pass a new array reference to SelectControl avoiding the need to re-render it.

Reply:
This logic can't be moved to a selector because it depends on the `image` prop. I mean we can still move it to a selector and name it something like `getImageSizeOptionsForImage` but I personally don't think it's a great idea to expose such specific selectors.

Reply:
Would it be possible to rename `availableImageSizes` to something more... contextual? Available sounds somewhat hard to understand as different sizes are available for different images (depending on when they were uploaded, original image size, etc.). Perhaps `sizeNameLabels` or `displayableImageSizes`?